### PR TITLE
VARIANT type for x64 and IOleInPlaceActiveObject are introduced

### PR DESCRIPTION
**Pull request by ryujimiya**
(1) Fix VARIANT type for Windows10 (x64) which was invalid at Windows 64-bit Windows
(2) IOleInPlaceActiveObject for Accelerator Keys are introduced
Thank you

### CODE REVIEW

File: oleaut32.go [hunk 1]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
 	return BSTRToString(value.bstrVal)
 }
 
+func (v *VARIANT) LVal() int32 {
```
Comment (lxn) [inline oleaut32.go]: This should be called `MustLong` and use `panic` instead of `log.Fatal` to allow recovery.

File: oleaut32.go [hunk 2]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	value, err := v.GetLVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetLVal() (int32, error) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `Long`.

File: oleaut32.go [hunk 3]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	if v.Vt != VT_I4 {
+		return 0, fmt.Errorf("Error: LVal() v.Vt !=  VT_I4, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_I4)(unsafe.Pointer(v))
+	return p.lVal, nil
+}
+
+func (v *VARIANT) SetLVal(value int32) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `SetLong`.

File: oleaut32.go [hunk 4]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	v.Vt = VT_I4
+	p := (*VAR_I4)(unsafe.Pointer(v))
+	p.lVal = value
+}
+
+func (v *VARIANT) ULVal() uint32 {
```
Comment (lxn) [inline oleaut32.go]: This should be called `MustULong` and use `panic` instead of `log.Fatal` to allow recovery.

File: oleaut32.go [hunk 5]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	value, err := v.GetULVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetULVal() (uint32, error) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `ULong`.

File: oleaut32.go [hunk 6]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	if v.Vt != VT_UI4 {
+		return 0, fmt.Errorf("Error: LVal() v.Vt !=  VT_UI4, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_UI4)(unsafe.Pointer(v))
+	return p.ulVal, nil
+}
+
+func (v *VARIANT) SetULVal(value uint32) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `SetULong`.

File: oleaut32.go [hunk 7]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	v.Vt = VT_UI4
+	p := (*VAR_UI4)(unsafe.Pointer(v))
+	p.ulVal = value
+}
+
+func (v *VARIANT) BoolVal() VARIANT_BOOL {
```
Comment (lxn) [inline oleaut32.go]: This should be called `MustBool` and use `panic` instead of `log.Fatal` to allow recovery.

File: oleaut32.go [hunk 8]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	value, err := v.GetBoolVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetBoolVal() (VARIANT_BOOL, error) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `Bool`.

File: oleaut32.go [hunk 9]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	if v.Vt != VT_BOOL {
+		return VARIANT_FALSE, fmt.Errorf("Error: BoolVal() v.Vt !=  VT_BOOL, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_BOOL)(unsafe.Pointer(v))
+	return p.boolVal, nil
+}
+
+func (v *VARIANT) SetBoolVal(value VARIANT_BOOL) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `SetBool`.

File: oleaut32.go [hunk 10]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	v.Vt = VT_BOOL
+	p := (*VAR_BOOL)(unsafe.Pointer(v))
+	p.boolVal = value
+}
+
+func (v *VARIANT) BstrVal() *uint16 {
```
Comment (lxn) [inline oleaut32.go]: This should be called `MustBSTR` and use `panic` instead of `log.Fatal` to allow recovery.

File: oleaut32.go [hunk 11]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	value, err := v.GetBstrVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetBstrVal() (*uint16, error) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `BSTR`.

File: oleaut32.go [hunk 12]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	if v.Vt != VT_BSTR {
+		return nil, fmt.Errorf("Error: BstrVal() v.Vt !=  VT_BSTR, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_BSTR)(unsafe.Pointer(v))
+	return p.bstrVal, nil
+}
+
+func (v *VARIANT) SetBstrVal(value *uint16) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `SetBSTR`.

File: oleaut32.go [hunk 13]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	v.Vt = VT_BSTR
+	p := (*VAR_BSTR)(unsafe.Pointer(v))
+	p.bstrVal = value
+}
+
+func (v *VARIANT) PDispVal() *IDispatch {
```
Comment (lxn) [inline oleaut32.go]: This should be called `MustPDispatch` and use `panic` instead of `log.Fatal` to allow recovery.

File: oleaut32.go [hunk 14]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	value, err := v.GetPDispVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetPDispVal() (*IDispatch, error) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `PDispatch`.

File: oleaut32.go [hunk 15]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	if v.Vt != VT_DISPATCH {
+		return nil, fmt.Errorf("Error: PDispVal() v.Vt !=  VT_DISPATCH, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_PDISP)(unsafe.Pointer(v))
+	return p.pdispVal, nil
+}
+
+func (v *VARIANT) SetPDispVal(value *IDispatch) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `SetPDispatch`.

File: oleaut32.go [hunk 16]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	v.Vt = VT_DISPATCH
+	p := (*VAR_PDISP)(unsafe.Pointer(v))
+	p.pdispVal = value
+}
+
+func (v *VARIANT) PVarVal() *VARIANT {
```
Comment (lxn) [inline oleaut32.go]: This should be called `MustPVariant` and use `panic` instead of `log.Fatal` to allow recovery.

File: oleaut32.go [hunk 17]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	value, err := v.GetPVarVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetPVarVal() (*VARIANT, error) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `PVariant`.

File: oleaut32.go [hunk 18]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	if v.Vt != VT_BYREF|VT_VARIANT {
+		return nil, fmt.Errorf("Error: PVarVal() v.Vt !=  VT_BYREF|VT_VARIANT, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_PVAR)(unsafe.Pointer(v))
+	return p.pvarVal, nil
+}
+
+func (v *VARIANT) SetPVarVal(value *VARIANT) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `SetPVariant`.

File: oleaut32.go [hunk 19]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	v.Vt = VT_BYREF | VT_VARIANT
+	p := (*VAR_PVAR)(unsafe.Pointer(v))
+	p.pvarVal = value
+}
+
+func (v *VARIANT) PBoolVal() *VARIANT_BOOL {
```
Comment (lxn) [inline oleaut32.go]: This should be called `MustPBool` and use `panic` instead of `log.Fatal` to allow recovery.

File: oleaut32.go [hunk 20]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	value, err := v.GetPBoolVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetPBoolVal() (*VARIANT_BOOL, error) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `PBool`.

File: oleaut32.go [hunk 21]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	if v.Vt != VT_BYREF|VT_BOOL {
+		return nil, fmt.Errorf("Error: PBoolVal() v.Vt !=  VT_BYREF|VT_BOOL, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_PBOOL)(unsafe.Pointer(v))
+	return p.pboolVal, nil
+}
+
+func (v *VARIANT) SetPBoolVal(value *VARIANT_BOOL) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `SetPBool`.

File: oleaut32.go [hunk 22]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	v.Vt = VT_BYREF | VT_BOOL
+	p := (*VAR_PBOOL)(unsafe.Pointer(v))
+	p.pboolVal = value
+}
+
+func (v *VARIANT) PPDispVal() **IDispatch {
```
Comment (lxn) [inline oleaut32.go]: This should be called `MustPPDispatch` and use `panic` instead of `log.Fatal` to allow recovery.

File: oleaut32.go [hunk 23]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	value, err := v.GetPPDispVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetPPDispVal() (**IDispatch, error) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `PPDispatch`.

File: oleaut32.go [hunk 24]
```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
...
+	if v.Vt != VT_BYREF|VT_DISPATCH {
+		return nil, fmt.Errorf("PPDispVal() v.Vt !=  VT_BYREF|VT_DISPATCH, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_PPDISP)(unsafe.Pointer(v))
+	return p.ppdispVal, nil
+}
+
+func (v *VARIANT) SetPPDispVal(value **IDispatch) {
```
Comment (lxn) [inline oleaut32.go]: This should be called `SetPPDispatch`.

**Comment by ryujimiya** [comment]
I made the changes following your review.
  The VARIANT-type method names are changed: e.g. LVal() to MustLong(). etc
https://github.com/lxn/win/pull/48/commits/b823cb1531857f4713b10f9e12074d7e3af78835
Thank you.

**Comment by ryujimiya** [comment]
Thank you for reviewing.
I've modified. Could you review the following?
https://github.com/lxn/win/pull/48/commits/7fd4c80abc743dac2b5cef7f172c2163bda4b092

**Comment by lxn** [comment]
Thank you for your nice contribution!

**Pull request merged**